azurelinuxagent/common/protocol/wire.py

```diff
@@ -913,47 +913,41 @@ def _call_hostplugin_with_container_check(self, host_func):
+            # or raising ResourceGone or InvalidContainer exceptions.
+            if not ret:
+                logger.periodic_info(logger.EVERY_HOUR, "[PERIODIC] Request failed using the direct channel, "
+                                                        "switching to host plugin.")
+        except (ResourceGoneError, InvalidContainerError) as e: # pylint: disable=C0103
+            logger.periodic_info(logger.EVERY_HOUR, "[PERIODIC] Request failed using the direct channel, "
+                                                    "switching to host plugin. Error: {0}".format(ustr(e)))
 
-        # NOTE: direct_func and host_func are passed as lambdas. Be careful about capturing goal state data in them as
-        # they will not be refreshed even if a goal state refresh is called before retrying the host_func.
+        return ret
 
-        if not HostPluginProtocol.is_default_channel():
-            ret = None
-            try:
-                ret = direct_func()
+    def send_request_using_appropriate_channel(self, direct_func, host_func):
+        # A wrapper method for all function calls that send HTTP requests. The first attempt is always via the
+        # host plugin channel. If, for whatever reason, the request failed, we fall back to using the direct channel.
 
-                # Different direct channel functions report failure in different ways: by returning None, False,
-                # or raising ResourceGone or InvalidContainer exceptions.
-                if not ret:
-                    logger.periodic_info(logger.EVERY_HOUR, "[PERIODIC] Request failed using the direct channel, "
-                                                            "switching to host plugin.")
-            except (ResourceGoneError, InvalidContainerError) as e: # pylint: disable=C0103
-                logger.periodic_info(logger.EVERY_HOUR, "[PERIODIC] Request failed using the direct channel, "
-                                                        "switching to host plugin. Error: {0}".format(ustr(e)))
+        # When using the host plugin channel, there is a retry mechanism. On specific failures that indicate
+        # a stale goal state (such as resource gone or invalid container parameter),
+        # refresh the goal state and try again. If failed, raise the exception.
 
-            if ret:
-                return ret
-        else:
-            logger.periodic_info(logger.EVERY_HALF_DAY, "[PERIODIC] Using host plugin as default channel.")
+        # NOTE: direct_func and host_func are passed as lambdas. Be careful about capturing goal state data in them as
+        # they will not be refreshed even if a goal state refresh is called before retrying the host_func.
 
-        ret = self._call_hostplugin_with_container_check(host_func)
+        ret = None
+        try:
+            ret = self._call_hostplugin_with_container_check(host_func)
+        except Exception:
+            pass
```

Comment:
In the `_call_hostplugin_with_container_check` function, we only catch `(ResourceGoneError, InvalidContainerError)` and log/reraise those. Incase its some other random exception, this would not even log it. We should log the errors incase the exception is not in `(ResourceGoneError, InvalidContainerError)`